chore(tutorial):Add DatoCMS/Snipcart tutorial (#22561)
gatsbyjs/gatsby: docs/tutorial/e-commerce-with-datocms-and-snipcart/index.md

- [Gatsby command line interface (CLI)](https://www.gatsbyjs.org/tutorial/part-zero/#using-the-gatsby-cli): run Gatsby commands on your computer

## Provisioning Your Site on Gatsby Cloud

Begin with Gatsby Cloud. Once you have an account, you can log in and follow these instructions:

- Select the "Create new site" option.

- Since you're starting from scratch, choose "I don't have a Gatsby site yet" on the next screen.

- Select the starter labeled "Snipcart with DatoCMS" from the list of ready-made starters (Gatsby sites that you can copy).

At this point, you'll need to come up with a name for your project. This won't be customer-facing. This name will identify your site in Gatsby Cloud and your repository (repo) on GitHub. For now, associate this site with your personal GitHub account. If you're building this site for your company, you can always transfer the project over to a [GitHub organization](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/about-organizations) later.

![Gatsby menu where you can create a new site asks What should we call your new project and choose a destination](/images/gatsby-new-site.png)

> ### Connect site with DatoCMS

Author: ```suggestion
### Connecting to DatoCMS
```

Authorize Gatsby to connect with your DatoCMS account and hit the "Start my site" button. Once this is done, you should see a link to the corresponding project in DatoCMS and repository on GitHub, both of which have been created for you.

![Log into DatoCMS overlaid on top of the Gatsby UI](/images/dato-log-in.png)

> This automatic provisioning of your website is one of Gatsby Cloud's strengths. Integrations with content management systems like this make it possible for you to spin up and begin publishing content to new sites without ever touching any code.

![celebratory screen titled successfully set up Gatsby provisioned Snipcart!](/images/successfully-set-up.png)

## Exploring the DatoCMS Project

After sucessfully linking the DatoCMS you can click on "Start my site" and
follow the link in Gatsby Cloud to go straight to your project in DatoCMS.

Click on "Enter project" and select "Products" from the secondary navigation menu. You'll see a number of products already defined for you. You can delete or edit these as you'd like.

![edit product screen featuring a single sock product called street faces](/images/edit-product.png)

> ### Editing models in DataCMS

Author: ```suggestion
### Editing models in DatoCMS
```

### Editing in GitHub

> If you're comfortable writing code in a text editor like [Visual Studio Code](https://code.visualstudio.com/) and using the command line, you can skip this section!

You can make small changes to the code from your browser by editing files directly in GitHub. Say, for example, that you always want to display prices in some other currency besides euros. To switch the symbols, head to your repository's page on GitHub and navigate to `src/pages/index.js`.

![github menu shows the current branch and file location as well as the first few lines of code in this file. Also includes view options raw, blame, and history](/images/github-menu.png)

Select "Edit this file" (the pencil icon) from the menu. Where `product.price` is being displayed, you'll need to update the currency symbol.

```jsx:title=index.js

> <div className="Product__price">${product.price}</div>

Author: I kept it this way because that's what it looks like when you use the Snipcart starter.

```suggestion
<div className="Product__price">
  ${product.price}
</div>
```

You've made a change to your master branch, which means Gatsby will build a new version of your site!

The rest of this process is described in the context of using a text editor to make code changes but remember that you can make those same changes in GitHub. One big advantage to developing locally is that you can see your changes reflected in the browser without having to wait for your entire site to rebuild.

### Editing Locally

> You can edit your site's code locally, by follwing these steps:

Author: ```suggestion
You can edit your site's code locally by following these steps:
```

> - Clone the project by Selecting the "Clone of download" button on your repository's main page will display a link for you to copy.

Author: ```suggestion
- Clone the project by Selecting the "Clone or download" button on your repository's main page, which will display a link for you to copy.
```

- Copy the project to your computer and install its dependencies, other JavaScript projects that your site relies on to work.

> The above is achieved by runing this command after copying the link on GitHub.

Author: ```suggestion
The above is achieved by running these commands after copying the link on GitHub.
```

```shell
git clone your-repo-link

> cd repo name

Author: ```suggestion
cd your-repo-name
```

![sample shop has a bright blue, pink, and green gradient as a background. Stacking rings, rhodochrosite ring, and statement earrings have been added as products](/images/sample-with-products.png)

> ## Modifying layout

Author: ```suggestion
## Modifying styles
```
